From 0ca7040f20998089d7004660ec819d4fef822fdf Mon Sep 17 00:00:00 2001
From: Johannes Berg <johannes.berg@intel.com>
Date: Sun, 2 Aug 2026 17:33:39 +0200
Subject: [PATCH 09809/15333] Revert "wifi: mac80211: don't encrypt pre-auth
 (ETH_P_PREAUTH) frames"

This reverts commit dd406779999fa2065ec6b7c4f80906b727041d2c.

Never encrypting the frames broke a number of tests that do
additional pre-authentication while already connected, and
then the frames didn't go out correctly. Whatever this was
intended to fix, this wasn't the right fix.

Fixes: dd406779999f ("wifi: mac80211: don't encrypt pre-auth (ETH_P_PREAUTH) frames")
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 net/mac80211/tx.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index 0cf5f6ec75e6..14311aab7d18 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -557,9 +557,6 @@ ieee80211_tx_h_check_control_port_protocol(struct ieee80211_tx_data *tx)
 		info->flags |= IEEE80211_TX_CTL_USE_MINRATE;
 	}
 
-	if (tx->skb->protocol == htons(ETH_P_PREAUTH))
-		info->flags |= IEEE80211_TX_INTFL_DONT_ENCRYPT;
-
 	return TX_CONTINUE;
 }
 
-- 
2.55.0

